code/modules/mob/living/carbon/human/species/species.dm

```diff
@@ -155,7 +159,13 @@
 
 	for(var/organ in has_organ)
 		var/organ_type = has_organ[organ]
+		//world << "[organ_type]"//debug print
 		H.internal_organs_by_name[organ] = new organ_type(H,1)
+		H.internal_organs += new organ_type
```

Comment:
I suspect that this line may be resulting in the "double the organs, double the fun" shenanigans that plagued you. You probably want to replace this with:
`
var/obj/item/organ/internal/O = new organ_type(H, 1)
O.insert(H)

You should also remove the internal_organs_by_name line, and add handling to organ/internal.insert() to populate that list, so you don't get any organ weirdness after reinsertion.